src/composables/useOnboard.ts

```diff
@@ -0,0 +1,27 @@
+import { inject, Ref } from 'vue';
+// @ts-ignore
+import Web3 from 'web3';
+import { API } from 'bnc-onboard/dist/src/interfaces';
+
+export type Web3State = {
+  onboardInstance: Ref<null | API>;
+  web3Instance: Ref<null | Web3>;
+};
+
+export const Web3ProviderSymbol = Symbol('WEB3_PROVIDER');
+
+export default function useOnboard() {
+  const web3State = inject<Web3State>(Web3ProviderSymbol);
```

Comment:
Maybe we could just add this to `useBlocknative` since we're already doing a similar thing there for notify.js? It's also where we have access to the blocknative sdk so it kind of is a wrapper for all the blocknative plugins.